routes.go

```diff
@@ -112,19 +114,40 @@ func Raw(c *gin.Context) {
 			contents := raw["contents"].([]interface{}) // Other MITM
 			for _, v := range contents {
 				entry := v.(map[string]interface{})
-				protoData = append(protoData, InboundRawData{
-					Base64Data: entry["payload"].(string),
-					Method:     int(entry["type"].(float64)),
-					HaveAr: func() *bool {
-						if v := entry["have_ar"]; v != nil {
-							res, ok := v.(bool)
-							if ok {
+				// Atlas, GC and GDS support
+				if userAgent[:13] == "Pokemod Atlas" || userAgent[:10] == "PokmonGO/0" {
```

Comment:
I think this will fail if user agent is shorter or missing than 13 chars. Should also check for length for safety.

Reply:
Thanks! I'll add check later